src/v/cloud_storage/async_manifest_view.h

```diff
@@ -34,9 +34,28 @@
 
 namespace cloud_storage {
 
+struct timequery_params {
+    explicit timequery_params(
+      model::timestamp ts, kafka::offset o = kafka::offset{})
+      : ts(ts)
+      , start_offset(o) {}
+
+    // Timestamp to search for.
+    model::timestamp ts{};
+
+    // Optional inclusive lower bound on the Kafka offset.
+    kafka::offset start_offset{};
```

Comment:
I'm wondering if this part is needed: why clamp this inside the cloud storage layer?  If the cloud storage timequery returns an offset behind the partition's start kafka offset, I would have thought we could have the Kafka layer advance to the start offset.

Reply:
:thinking: 

That's true, although we promise the user very little when it comes to out of order timestamps, so I'd recommend going with whatever makes implementation cleanest/simplest.